extensions/amp-list/0.1/amp-list.js

```diff
@@ -204,60 +214,58 @@ export class AmpList extends AMP.BaseElement {
       // Remove any previous items before the reload
       removeChildren(dev().assertElement(this.container_));
     }
-
-    const itemsExpr = this.element.getAttribute('items') || 'items';
-    return this.fetch_(itemsExpr).then(items => {
-      if (this.element.hasAttribute('single-item')) {
-        user().assert(typeof items !== 'undefined' ,
-            'Response must contain an array or object at "%s". %s',
+    if (this.viewer_.canRenderTemplates()) {
```

Comment:
Let's make this code block more readable by creating more private helper methods.